src/styles/blocks/_catalog.scss

```diff
@@ -0,0 +1,27 @@
+.catalog {
+  display: grid;
+  grid-auto-flow: row;
+  grid-template-columns: repeat(4, $card-width);
+  grid-template-rows: repeat(auto-fill, $card-height);
+
+  justify-content: center;
+
+  gap: 48px;
+
+  &__card {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: $card-width;
+    height: $card-height;
+    border: 1px solid #ccc;
+    border-radius: 5px;
+    margin-bottom: 48px;
+  }
+}
+
+@media (max-width: 1024px) {
+  .catalog {
+    grid-template-columns: repeat(auto-fit, $card-width);
+  }
+}
```

Comment:
Read conditions again. You need to add a several @media queries
<img width="429" alt="CleanShot 2022-08-30 at 15 36 06@2x" src="https://user-images.githubusercontent.com/103505670/187438123-73773704-36a1-4cbb-864f-ddab794ab106.png">

If i stretch your page it takes wrong number of cards, 4 cards must be from 1024px and so on for 768px, 488px...
![CleanShot 2022-08-30 at 15 37 36@2x](https://user-images.githubusercontent.com/103505670/187438753-8bcc2aae-7617-414b-aeee-b6b7590349a4.png)

Maybe you better start from mobile version (>488px) and add @media queries for next screen sizes